blog/blog-library/src/main/java/org/silverpeas/components/blog/service/DefaultBlogService.java

```diff
@@ -296,191 +293,146 @@ private Connection openConnection() {
-
-      return post;
-    } catch (Exception e) {
-      throw new BlogRuntimeException(
-          failureOnGetting(POST + " associated to publication", publication.getId()), e);
+  private List<PostDetail> toPosts(final String instanceId,
+    final Collection<PublicationDetail> publications) {
+    if (!publications.isEmpty()) {
+      final Map<Location, Category> categoryByAllNodePKs = nodeService.getAllNodes(
+              new NodePK(null, instanceId))
+          .stream()
+          .collect(toMap(n -> new Location(n.getId(), instanceId), Category::new));
+      final Collection<String> pubIds = publications.stream()
+          .map(PublicationDetail::getPK)
+          .map(PublicationPK::getId)
+          .collect(toSet());
+      final Map<String, Category> allLocationsByPublicationIds =
+          publicationService.getAllLocationsByPublicationIds(pubIds)
+          .entrySet()
+          .stream()
+          .collect(toMap(Map.Entry::getKey, e -> categoryByAllNodePKs.get(e.getValue().get(0))));
+      final Map<String, Date> dateByEvents = getDateByEvents(pubIds);
+      final Map<String, Integer> commentsByPubIds =
+          getCommentService().getInstanceCommentsCountByResourceRefs(
+              PostDetail.getResourceType(), instanceId)
+          .entrySet()
+          .stream()
+          .collect(toMap(e -> e.getKey().getLocalId(), Map.Entry::getValue));
+      final Date defaultDate = new Date();
+      return publications.stream().map(p -> {
+        final String pubId = p.getPK().getId();
+        final Category cat = allLocationsByPublicationIds.get(pubId);
+        final Date dateEvent = dateByEvents.getOrDefault(pubId, defaultDate);
+        final int nbComments = commentsByPubIds.getOrDefault(pubId, 0);
+        final PostDetail post = new PostDetail(p, cat, nbComments, dateEvent);
+        //noinspection removal
+        post.setCreatorName(p.getCreator().getDisplayedName());
+        return post;
+      }).collect(toList());
     }
+    return List.of();
   }
 
   @Override
-  public Collection<PostDetail> getAllPosts(String instanceId) {
-    Collection<PostDetail> posts = new ArrayList<>();
+  public Collection<PostDetail> getLastPosts(final String instanceId, final BlogFilters filters) {
     try (Connection con = openConnection()) {
-      // rechercher les publications classée par date d'évènement
-      Collection<String> lastEvents = PostDAO.getAllEvents(con, instanceId);
-      Collection<PublicationDetail> publications =
-          publicationService.getAllPublications(instanceId);
-      for (String pubId : lastEvents) {
-        for (PublicationDetail publication : publications) {
-          if (publication.getPK()
-              .getId()
-              .equals(pubId)) {
-            posts.add(getPost(publication));
-          }
-        }
-      }
-      return posts;
+      // search publications sorted by event date
+      Collection<String> lastEventIds = PostDAO.getAllEvents(con, instanceId);
```

Comment:
Pfff, this method name was originally misnamed. In fact, what this method does is to get all the blog posts of the specified component instance ordered by their event date. So the method should be renamed to something like `getAllPostsOrderedByEventDate`.
And then the variable shouldn't be named lastEventIds but something like `postsOrderedByEventDate`.

My remark is also for all similar snippet of code written in the next methods below (this allow me to not repeat this comment)